lib/driver.js

```diff
@@ -292,12 +293,10 @@ class IosDriver extends BaseDriver {
       }
     }
 
-    await this.checkPreferences();
-
     await this.runSimReset();
     await this.isolateSimDevice();
-    await this.setLocale();
-    await this.setPreferences();
+    await this.setLocale(this.sim);
```

Comment:
since we're pulling `this.sim` out, it makes me wonder if we can go whole hog and remove all instances of it; do we really use it anywhere other than in functions called by `startSimulator`?